docs/proposals/apiman-integration.md

```diff
@@ -0,0 +1,132 @@
+# APIMan Integration
+
+## Abstract
+This proposes a design for integrating [APIMan](http://www.apiman.io/) with OpenShift in order to provide  micro-services governance for API service providers.
+
+## Motivation
+Providers of services may have the need to control who and how their services are consumed.  This may be realized as policies defined by providers to control a service
+in any number of ways (e.g security, throttling/quota, billing and metrics)<sup>[1](#r1)</sup>. 
+
+## Constraints and Assumptions
+### Authentication
+The OAuth token identifying a user in OpenShift will be utilized by APIMan to associate service policy to their identity.  This will ensure a user's identity is 
+ubiquitous across the cluster when interacting with OpenShift and the APIMan management console.
+
+### Deployment
+#### Communication
+APIMan Components (e.g. gateway, management interface) integrated with the cluster will utilize mutual TLS for internal communication.
+
+#### Storage
+APIMan is capable of using several backends (e.g. ElasticSearch, MySql) to store configuration, policy, and metrics.  The initial integration will utilize ElasticSearch with the goal of reusing the same ElasticSearch cluster that is deployed for aggregated logging.  It is assumed when the backend resides on the cluster, [Origin Aggregated logging](https://github.com/openshift/origin-aggregated-logging)<sup>[4](#r4)</sup> will be deployed before APIMan. 
```

Comment:
@smarterclayton @danmcp We have been discussing this decision on the integration services team and whether it makes sense to make apiman dependent upon logging.  We have specific concerns related to:

- Upgrading system components independently
- Performance considerations (e.g. high log volume hindering apiman access to its policies, disabled logging - see @bparees comments regarding enduser feedback)

It is my understanding the storage requirements are small since we will not have apiman's metrics capabilities enabled.  @KurtStam has stated we have other storage options which may be a smaller footprint then ES but it would introduce another dependency into OpenShift which @smarterclayton  has expressed reservations about.

Do we really want to tie APIMan to logging? Do we have a preferred storage alternative?